pkg/yurthub/tenant/tenant.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2022 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package tenant
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+
+	"github.com/openyurtio/openyurt/pkg/yurthub/util"
+)
+
+type Interface interface {
+	GetTenantNs() string
+
+	GetTenantToken() string
+
+	InformerSynced() cache.InformerSynced
+
+	SetSecret(sec *v1.Secret)
+}
+
+type tenantManager struct {
+	Informer cache.SharedIndexInformer
+
+	secretSynced cache.InformerSynced
+
+	TenantSecret *v1.Secret
+
+	TenantNs string
+}
+
+func New(orgs []string, factory informers.SharedInformerFactory) Interface {
+
+	tenantNs := util.ParseTenantNsFromOrgs(orgs)
+	klog.Infof("parse tenant ns: %s", tenantNs)
+	if tenantNs == "" {
+		return nil
+	}
+
+	tenantMgr := &tenantManager{TenantNs: tenantNs}
+
+	if factory != nil {
+		tenantMgr.Informer = factory.InformerFor(&v1.Secret{}, nil) //get registered secret informer
```

Comment:
factory.InformerFor(&v1.Secret{}, nil) --> sharedFactory.Core().V1().Secrets().Informer()

Reply:
ok, but i think it's more readable to use `sharedFactory.Core().V1().Secrets().Informer()` to get informer.